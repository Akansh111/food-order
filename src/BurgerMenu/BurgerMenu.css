.food-menu {
    height: 30rem;
    overflow-y: auto;
    display: flex;
}
.ag-grid {
    width: 54%;
}