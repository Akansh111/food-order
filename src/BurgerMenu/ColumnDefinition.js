
export const colDefs = [
    { 
        headerName: "Burger Name",
        field: "burgerName",
        filter: true,
        sortable: true,
        filterParams: { suppressMenu: true, suppressMiniFilter: true, filter: false  },
        headerCheckboxSelection: true,
          headerCheckboxSelectionFilteredOnly: true,
          checkboxSelection: true,
    },
    {
        headerName: "Category", 
        field: "category" ,
        filter: true,
        sortable: true,
        width: 150,
        filterParams: { suppressMiniFilter: true }

    },
    { 
        headerName: "Rating", 
        field: "rating", 
        filter: true,
        sortable: true,
        width: 120,
        filterParams: { suppressMiniFilter: true }

    },
    { 
        headerName: "Price (â‚¹)", 
        field: "price", 
        filter: true,
        sortable: true,
        width: 120,
        filterParams: { suppressMiniFilter: true }

    },
    { 
        headerName: "Discount (%)", 
        field: "discount", 
        filter: true,
        sortable: true,
        width: 120,
        filterParams: { suppressMiniFilter: true }

    }
];