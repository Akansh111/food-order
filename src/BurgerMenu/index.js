export * from './BurgerMenu';
export * from './ColumnDefinition';