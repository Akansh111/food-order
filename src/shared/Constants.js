export const url = 'https://demo8151081.mockable.io/getFoodData';
export const mainTitle = 'Lets BANG ON to your Favourite Burger';
export const orderNowTitle = 'ORDER NOW!!!';
export const menuMessage = 'Select the items from the Menu!';
export const successMessage = 'Wohoo, Your order will be delivered in 30 minutes!';
export const discountMessage = 'after applying discount';
export const orderSummaryTitle = 'ORDER SUMMARY';