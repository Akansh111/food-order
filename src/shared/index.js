export * from './Constants';
export * from './mockApiData';
