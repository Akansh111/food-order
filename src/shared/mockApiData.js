export const mockApiData = [
    {
        "burgerName": "Panner King's Melt",
        "rating": "★★★★★",
        "price": 129,
        "category": "Veg",
        "discount": 10
    },
    {
        "burgerName": "Veggie",
        "rating": "★★★",
        "price": 95,
        "category": "Veg",
        "discount": 5
    },
    {
        "burgerName": "Makani Supreme Veg",
        "rating": "★★★★★",
        "price": 79,
        "category": "Veg",
        "discount": 5
    },
    {
        "burgerName": "Crispy Veg",
        "rating": "★★★★",
        "price": 55,
        "category": "Veg",
        "discount": 5
    },
    {
        "burgerName": "Fiery Chicken",
        "rating": "★★★★",
        "price": 159,
        "category": "Non-Veg",
        "discount": 15
    },
    {
        "burgerName": "Chicken Chilli Cheese Melt",
        "rating": "★★★★★",
        "price": 129,
        "category": "Non-Veg",
        "discount": 10
    },
    {
        "burgerName": "Crispy Chicken",
        "rating": "★★★",
        "price": 75,
        "category": "Non-Veg",
        "discount": 5
    },
    {
        "burgerName": "Crispy Chicken Supreme",
        "rating": "★★★★★",
        "price": 85,
        "category": "Non-Veg",
        "discount": 5
    }
]