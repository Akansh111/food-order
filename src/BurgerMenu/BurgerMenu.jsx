import React from 'react';
import axios from 'axios';
import { AgGridReact } from 'ag-grid-react';
import { colDefs } from './ColumnDefinition';
import { OrderSummary } from '../OrderSummary';
import { url } from '../shared';

import './BurgerMenu.css';

export const BurgerMenu = () => {
    
    const [foodData, setFoodData] = React.useState([]);
    const [summaryData, setSummaryData] = React.useState([]);
    React.useEffect(() => {
        const fetchData = async () => {
            const result = await axios(url);
            setFoodData(result.data);
        };
        fetchData();
    }, []);

    const onSelectionChanged = (params) => {
        const selectedRows = params.api.getSelectedRows();
        setSummaryData(selectedRows);
    };
    
    return (
        <div className="food-menu ag-theme-balham"> 
            <div className="ag-grid">
                <AgGridReact
                    columnDefs={colDefs}
                    rowData={foodData}
                    floatingFilter={true}
                    rowSelection='multiple'
                    onSelectionChanged={onSelectionChanged}

                />
            </div>
            <OrderSummary
                summaryData = {summaryData}
            />
            
        </div>
    );
}