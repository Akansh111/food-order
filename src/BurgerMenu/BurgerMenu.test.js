import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { url, mockApiData } from '../shared'
import { BurgerMenu } from './BurgerMenu';

var axios = require("axios");
var MockAdapter = require("axios-mock-adapter");

describe('renders BurgerMenu', () => {

    it('renders', () => {
        const component = renderer.create(<BurgerMenu />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
  
  
    it('should render a proper table data', done => {
        const mock = new MockAdapter(axios);
        mock.onGet(url).reply(200, mockApiData);
        const component = shallow(<BurgerMenu />);
        setImmediate(() => {
            component.update();
            done();
        });
    });

});