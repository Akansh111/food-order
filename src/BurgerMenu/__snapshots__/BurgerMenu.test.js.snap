// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BurgerMenu renders 1`] = `
<div
  className="food-menu ag-theme-balham"
>
  <div
    className="ag-grid"
  >
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  </div>
  <div
    className="order-summary"
  >
    <div
      className="card-title"
    >
      ORDER SUMMARY
    </div>
    <br />
    <div
      className="summary-table"
    >
      <table
        className="table table-striped"
      >
        <tbody>
          <div>
            Select the items from the Menu!
          </div>
        </tbody>
      </table>
    </div>
    <div
      className="cart-total"
    >
      Total: â‚¹0.00
    </div>
    <button
      className="btn btn-primary btn-block btn-lg"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Confirm Order
    </button>
  </div>
</div>
`;
